api

```diff
@@ -351,6 +351,38 @@ app_to_pkgname() { #given an app-name, convert it to a unique, valid package-nam
   echo "pi-apps-$(echo "$app" | md5sum | cut -c1-8 | awk '{print $1}')"
 }
 
+add_external_repo() { # add an external apt repo and its gpg key
+  repopath="$1"
+  reponame="$2"
+  pubkeyurl="$3"
+
+  # make pi-apps key storage dir if it doesn't exist
+  if [ ! -d /tmp/pi-apps-apt-keys ]; then
+    mkdir -p /tmp/pi-apps-apt-keys || error "add_external_repo: failed to create pi-apps key storage dir."
+  fi
+
+  # check if .list file already exists
+  [ -f /etc/apt/sources.list.d/${reponame}.list ] && error "add_external_repo: /etc/apt/sources.list.d/${reponame}.list already exists."
+
+  # download gpg key from specified url
+  rm -f /tmp/pi-apps-apt-keys/${reponame}-keyring.gpg
+  wget -q "$pubkeyurl" -O /tmp/pi-apps-apt-keys/${reponame}-keyring.gpg || error "add_external_repo: download from specified pubkeyurl failed." && rm -f /tmp/pi-apps-apt-keys/${reponame}-keyring.gpg
+
+  # create .list file
+  echo "deb [signed-by=/tmp/pi-apps-apt-keys/$reponame-keyring.gpg] $repopath" | sudo tee /etc/apt/sources.list.d/${reponame}.list >/dev/null || error "add_external_repo: failed to create .list file"
```

Comment:
You seem to be treating files in /tmp as though they are persistent. They are not - every once in a while old contents will be cleared, and a reboot will clear it too.
GPG keys should be stored in a persistent, root-owned location.

Reply:
I changed the location from `/tmp` to `/opt/pi-apps-apt-keys`.

Reply:
@theofficialgman, where are system-wide GPG keys stored for things like apt repositories? Wouldn't it be better to store these keys in a standard location than a new one?

Reply:
> @theofficialgman, where are system-wide GPG keys usually stored for things like apt repositories? Wouldn't it be better to store these keys in a standard location than a new one?

It looks like `/usr/share/keyrings/`.. Don't quote me on that.. 😆

Reply:
Yup thats the correct location for all apt keyrings

Reply:
changed the location of the keys to `/usr/share/keyrings`.